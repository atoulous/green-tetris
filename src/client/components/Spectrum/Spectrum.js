import React from 'react';
import PropTypes from 'prop-types';
import Row from '../Row/Row';

import './Spectrum.scss';

const Spectrum = ({ spectrum }) => (
  <div className="spectrum">
    {spectrum.map((e, i) => (<Row key={i} row={e} />))}
  </div>
);

Spectrum.propTypes = {
  spectrum: PropTypes.array,
};

Spectrum.defaultProps = {
  spectrum: [],
};


export default Spectrum;
