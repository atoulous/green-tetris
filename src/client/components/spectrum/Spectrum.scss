.spectrum {
  display: flex;
  border: solid 1px black;
  height: 250px;
  width: 100%;
  align-items: stretch;
}

.spectrum .row {
  display: flex !important;
  flex-direction: column !important;
  flex: 1;
}

.spectrum .row .cell {
  border: solid gray 1px;
  width: 100%;
  flex: 1;
}
