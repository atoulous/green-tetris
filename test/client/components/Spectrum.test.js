import React from 'react';
import renderer from 'react-test-renderer';

import Spectrum from '../../../src/client/components/Spectrum/Spectrum';

it('renders ok', () => {
  const tree = renderer
    .create(<Spectrum spectrum={[12, 14, 3, 0, 0, 0, 8]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

