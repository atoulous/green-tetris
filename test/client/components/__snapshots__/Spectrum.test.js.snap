// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<div
  className="spectrum"
>
  <div
    className="row"
  >
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
  </div>
  <div
    className="row"
  >
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
  </div>
  <div
    className="row"
  >
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
  </div>
  <div
    className="row"
  >
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
  </div>
  <div
    className="row"
  >
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
  </div>
  <div
    className="row"
  >
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
  </div>
  <div
    className="row"
  >
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={Object {}}
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="cell"
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
  </div>
</div>
`;
